spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/ExpireSnapshotsSparkAction.java

```diff
@@ -67,6 +67,8 @@ public class ExpireSnapshotsSparkAction extends BaseSparkAction<ExpireSnapshotsS
 
   public static final String STREAM_RESULTS = "stream-results";
   public static final boolean STREAM_RESULTS_DEFAULT = false;
+  public static final String DELETE_FILES = "delete-files";
```

Comment:
I think it's better to handle this in a separate PR

Reply:
Can you elaborate which parts should be separated?

Reply:
Do you mean separate change to Spark action from core/api?

Reply:
introducing `deleted_snapshots_count` should be separated from introducing the `delete-files` flag as they don't seem to be related. For example, it's nice to have `deleted_snapshots_count` in the output, but that is an independent feature from the new flag

Reply:
sure, although deletedSnapshotsCount should equal to deletedManifestListsCount if deleting files

Reply:
@nastra I've created #9605 for `deleted_snapshots_count` and will rebase this PR later.